### DISCUSSION TITLE
Obtain torque of each joints and torque control

### DISCUSSION THREAD

**Discussion by hygradme** [MrAsana/AMBER_B1_ROS2#5] [Q&A]
It looks so far I cannot find the way to get the torque of each joint.
https://github.com/MrAsana/AMBER_B1_ROS2/wiki/SDK-&-API---UDP-Ethernet-Protocol--for-controlling-&-programing

How to get  torque of each joints?

Also, I'm interested in Impedance Control as is mentioned in the Kickstarter description page, 
https://www.kickstarter.com/projects/amberobotics/a-new-generation-of-7-axis-robot-arm-for-creative-tasks/description

How to use this torque control mode?

**Comment by MrAsana** [comment]
As you know, B1 doesn't have torque sensor, Impedance Control  feature is only for research in labs. If you want to use this feature , you have to do lots of work to  estimate the interaction force. The estimated interaction force will be used in the impedance control algorithms.  
The motor drive and controller support current loop, but it's not enable by default as it's still under developped, not stable to implement. 
If you're doing some research on this subject, we'd like to help you to keep going. But if you want to implement a dedicated task by B1, the position and speed loop control are the way to try.

**Reply by hygradme** [reply]
OK I see.
So how can I get current of each joint?
